test/libyul/YulOptimizerTestCommon.cpp

```diff
@@ -194,6 +194,8 @@ YulOptimizerTestCommon::YulOptimizerTestCommon(
 		}},
 		{"fullSimplify", [&]() {
 			disambiguate();
+			FunctionGrouper::run(*m_context, *m_ast);
+			BlockFlattener::run(*m_context, *m_ast);
```

Comment:
If this works, I actually think only
```
BlockFlattener::run(*m_context, *m_ast);
```
alone should in fact also work.

Reply:
BlockFlattener throws if FunctionGrouper has not been run before

https://github.com/ethereum/solidity/blob/88208707c2daec7dbc65041b2c555c81f38ea266/libyul/optimiser/BlockFlattener.cpp#L55